File: 2106-find-greatest-common-divisor-of-array/find-greatest-common-divisor-of-array.py
class Solution(object):
    def findGCD(self, nums):
        nums=sorted(nums)
        l=len(nums)-1
        a=nums[0]
        b=nums[l]
        ans=1
        for i in range(1,min(a,b)+1):
            if a%i==0 and b%i==0:
                ans=i
        return ans
        

File: 2106-find-greatest-common-divisor-of-array/find-greatest-common-divisor-of-array.java
import java.util.Arrays;
class Solution {
    public int findGCD(int[] nums) {
        int l=nums.length-1;
        Arrays.sort(nums);
        int a=nums[0];
        int b=nums[l];
        int ans=1;
        for(int i=1;i<=Math.min(a,b);i++){
            if (a%i==0 && b%i==0){
                ans=i;
            }
        }
        return ans;
    }
    
}